tests/solana_tests/rational.rs

```diff
@@ -208,4 +208,32 @@ fn rational() {
             },
         ]
     );
+
+    let mut vm = build_solidity(
+        r#"
+        contract foo {
+            function test1() public returns (bool) {
+                bool x = 1.1 != 1.0;
+                assert(x);
+                assert(1.1 == 1.1);
+                assert(1.2 >= 1.1);
+                assert(1.1 >= 1.1);
+                assert(1.1 <= 1.2);
+                assert(1.1 <= 1.1);
+                assert(1.2 > 1.1);
+                assert(1.1 < 1.2);
+                assert(1.1 > 1.2 || 1.1 < 1.2);
+                assert(1.2 > 1.1 && 1.1 < 1.2);
+                assert(1.2 > 1.1 ? 1.2 < 1.3 : 1.2 > 1.3);
+
+                return 1.1 > 1.0 && 1.1 >= 1.1;
+            }
```

Comment:
I added a `panic!` to every branch you coded in `eval_const_bool`, and removed the ones that indeed panicked. Your test function does not passes through `Expression::NotEqual`, `Expression::Or`, `Expression::And`, `Expression::ConditionalOperator` and `Expression::Not`, because the panics added there had no effect when building this function.

Reply:
In addition, panics at `Expression::NotEqual`, `Expression::Or`, `Expression::ConditionalOperator` did not produce a single failure when running `cargo test --workspace`

Reply:
`Expression::NotEqual` is interesting. We just never construct it, we build `Expression::Not { Expression::Equal }`.

I think we really should be constructing an ast which represents the source code much better. But that is something for another day.

Reply:
The new code does not have this change.